src/CareTogether.Core/Extensions.cs

```diff
@@ -15,10 +15,16 @@ public static ImmutableList<T> With<T>(this ImmutableList<T> list, T valueToUpda
             return list.Select(x => predicate(x) ? valueToUpdate : x).ToImmutableList();
         }
 
-        public static ImmutableList<T> UpdateSingle<T>(this ImmutableList<T> list, Func<T, bool> predicate,
-            Func<T, T> selector)
+        public static ImmutableList<T> AddOrReplace<T>(this ImmutableList<T> list, Func<T, bool> predicate,

```

Comment:
Make this a separate method, leaving `UpdateSingle` (and its uses in _ReferralModel.cs and anywhere else) intact.